pycbc/io/live.py

```diff
@@ -288,11 +295,46 @@ def upload(self, fname, gracedb_server=None, testing=True,
                 snr_series_fname = fname.replace('.xml.gz', '.hdf')
             else:
                 snr_series_fname = fname.replace('.xml', '.hdf')
+            snr_series_plot_fname = snr_series_fname.replace('.hdf',
+                                                             '_snr.png')
+            psd_series_plot_fname = snr_series_fname.replace('.hdf', 
+                                                             '_psd.png')
+            pylab.figure()
             for ifo in self.snr_series:
-                self.snr_series[ifo].save(snr_series_fname,
-                                          group='%s/snr' % ifo)
-                self.psds[ifo].save(snr_series_fname,
-                                    group='%s/psd' % ifo)
+                curr_snrs = self.snr_series[ifo]
+                curr_snrs.save(snr_series_fname, group='%s/snr' % ifo)
+                pylab.plot(curr_snrs.sample_times, abs(curr_snrs), 
+                           c=ifo_color(ifo), label=ifo)
+                if ifo in self.ifos:
+                    snr = self.coinc_results['foreground/%s/%s' %
+                                             (ifo, 'snr')]
+                    endt = self.coinc_results['foreground/%s/%s' %
+                                              (ifo, 'end_time')]
+                    pylab.plot([endt], [snr], c=ifo_color(ifo), marker='x')
+
+            pylab.legend()
+            pylab.xlabel('GPS time (s)')
+            pylab.ylabel('SNR')
+            pylab.savefig(snr_series_plot_fname)
+            pylab.close()
+
+            pylab.figure()
+            for ifo in self.snr_series:
+                # Undo dynamic range factor
+                curr_psd = FrequencySeries(self.psds[ifo],
+                                           dtype=numpy.float64)
+                curr_psd /= pycbc.DYN_RANGE_FAC ** 2.0
+                curr_psd.save(snr_series_fname, group='%s/psd' % ifo)
+                # Can't plot log(0) so start from point 1
+                pylab.loglog(curr_psd.sample_frequencies[1:],
+                             curr_psd[1:], c=ifo_color(ifo), label=ifo)
```

Comment:
Just my personal taste, but I would rather plot the ASD.

Reply:
Done